### PR TITLE
chore(CPlusPlus): added breadth first search

### PR DESCRIPTION
**Pull request by hkumar0132**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**

**Other information:**

### CODE REVIEW

File: algorithms/CPlusPlus/Graphs/breadth-first-search.cpp [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+int main()
+{
+    // Create a graph given in the above diagram
+    Graph g(4);
+    g.addEdge(0, 1);
+    g.addEdge(0, 2);
+    g.addEdge(1, 2);
+    g.addEdge(2, 0);
+    g.addEdge(2, 3);
+    g.addEdge(3, 3);
```
Comment (aayushjain7) [inline algorithms/CPlusPlus/Graphs/breadth-first-search.cpp:88]: Convert it to user-defined inputs if possible.

File: algorithms/CPlusPlus/Graphs/breadth-first-search.cpp [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+ 
+    cout << "Following is Breadth First Traversal "
+         << "(starting from vertex 2) \n";
+    g.BFS(2);
+ 
+    return 0;
+}
```
Comment (aayushjain7) [inline algorithms/CPlusPlus/Graphs/breadth-first-search.cpp:95]: Also, add a test case with time complexity

**Review by aayushjain7 — CHANGES_REQUESTED** [review]
all add it into readme.MD file is missing

File: algorithms/CPlusPlus/Graphs/breadth-first-search.cpp (hunk 2 above)
Comment (hkumar0132) [reply algorithms/CPlusPlus/Graphs/breadth-first-search.cpp:95]: Working on it, thanks

**Comment by hkumar0132** [comment]
I have made the necessary changes

**Review by aayushjain7 — APPROVED** [review]
Good work! 👍🏼

**Pull request merged**